miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/ArrayModelControllerST.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.bbsrc.tgac.miso.webapp.springtest;
+
+import org.junit.Test;
+
+import uk.ac.bbsrc.tgac.miso.core.data.ArrayModel;
+import uk.ac.bbsrc.tgac.miso.core.util.MapBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ArrayModelControllerST extends AbstractST {
+
+  private static final String CONTROLLER_BASE = "/arraymodel";
+  private static final Class<ArrayModel> entityClass = ArrayModel.class;
+
+  private List<HashMap<String, Object>> makeObjectMaps(List<Integer> ids) {
+    List<HashMap<String, Object>> mappedProperties = new ArrayList<HashMap<String, Object>>();
+
+    for (int i = 0; i < ids.size(); i++) {
+      MapBuilder<String, Object> currMap = new MapBuilder<String, Object>();
+      ArrayModel arrayModel = currentSession().get(entityClass, ids.get(i));
+      currMap.put("id", ids.get(i)).put("alias", arrayModel.getAlias());
+      mappedProperties.add((HashMap) currMap.build()); // safe cast since MapBuilder using HashMap
+    }
```

Comment:
This is why I'm not seeing the purpose of this abstraction. Building the properties map seems like at least as much effort as writing your assertions here for each field would be.